### ISSUE TITLE
Rule suggestion: v-model cannot be used on a prop

### ISSUE THREAD

**Issue by BARNZ** [vuejs/eslint-plugin-vue#2035]
**Please describe what the rule should do:**

This rule should check for the recently added vue compiler warning about using props directly with `v-model`

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
    <input v-model="someProp" />
</template>

<script setup>
    defineProps({ someProp: String })
</script>
```

**Additional context**

The above example now throws an error when compiling with vite: 

`[vite:vue] v-model cannot be used on a prop, because local prop bindings are not writable.`

https://github.com/vuejs/core/issues/5584#issuecomment-1309694870

**Comment by ota-meshi** [comment]
Thank you for posting the issue.
You can already check by using the `vue/no-mutating-props` rule.
https://eslint.vuejs.org/rules/no-mutating-props.html
